### PR TITLE
Include tests for Rx, Ry, Rz, U2, U3, Hz & FSim

### LINKED ISSUE #35: Test numerical correctness of parametric gates
Currently, only `Rxx`, `Ryy` and `Rzz` are tested.

### PR DESCRIPTION
**Pull request by Todorbsc**
Closes #35

### CODE REVIEW

**Comment by Todorbsc** [comment]
Just to verify, I would like you to review the definition of the `Hz` and `FSim` and compare it to the ones defined in Cirq. Probably, you did some equivalence as the one @mofeing told me about the RZ gate.

image

Cirq Hz:
image

Cirq FSim:
image

File: test/Array_test.jl [hunk 1]
```diff
@@ -160,6 +160,22 @@
...
+            @test begin
+                g = Rx(1, θ = π / 2)
+                Matrix(g) ≈ 1 / sqrt(2) * [1 -1im; -1im 1]
+            end
+
+            @test begin
+                g = Rx(1, θ = π)
+                Matrix(g) ≈ [0 -1im; -1im 0]
+            end
+        end
```
Comment (mofeing) [inline test/Array_test.jl:178]: You forgot to add a newline here.
```suggestion
        end

```

**Review by mofeing — APPROVED** [review]
Great thanks! I will revise `FSim` and `Hz` now.

File: test/Array_test.jl [hunk 2]
```diff
@@ -213,5 +263,134 @@
...
+                g = FSim(1, 2, θ = π, ϕ = π / 2)
+                Matrix(g) ≈ [1 0 0 0; 0 -1 -1im 0; 0 -1im -1 0; 0 0 0 1]
+            end
+
+            @test begin
+                g = FSim(1, 2, θ = π, ϕ = π)
+                Matrix(g) ≈ [1 0 0 0; 0 -1 0 0; 0 0 -1 0; 0 0 0 1]
+            end
+        end
+
```
Comment (jofrevalles) [inline test/Array_test.jl:394]:
```suggestion
```

Remove this empty line

**Review by jofrevalles — CHANGES_REQUESTED** [review]

**Review by jofrevalles — APPROVED** [review]

**Comment by mofeing** [comment]
The `Hz` gate seems correct. Check out that `cis(x) = exp(1im * x)`.
But `FSim` seems to be wrong... would you mind fixing it?

**Comment by Todorbsc** [comment]
> The `Hz` gate seems correct. Check out that `cis(x) = exp(1im * x)`. But `FSim` seems to be wrong... would you mind fixing it?

I fixed `FSim` according to the Cirq definition of the gate. Check it out.
Regarding the `Hz` gate, there is an opposite phase in the off-diagonal with respect to the Cirq definition while the diagonal is the same.

Cirq:
image

Our definition:
$$ \begin{bmatrix}
e^{i \theta / 2} cos(\theta / 2) & -i e^{-i (\theta / 2 - \phi)} sin(\theta / 2) \\
-i e^{-i (\theta / 2 + \phi)} sin(\theta / 2) & e^{i \theta / 2} cos(\theta / 2) \\
\end{bmatrix}, \theta = \pi t , \phi = \pi t$$

If this is not a problem, we can merge the PR.

**Review by mofeing — APPROVED** [review]
Great! Thanks!

**Pull request merged**